### PR TITLE
Implementation of "batch subscription" functionality

### PR DESCRIPTION
**Pull request by samatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
  - Update of ElementDefinitionBrowser to include context menu item and command to execute batch ParameterSubscriptions
  - Add IParameterSubscriptionBatchService and ParameterSubscriptionBatchService
  - Add ParameteterSubscriptionFilterSelectionDialogViewModel to select filters on the basis of which Parameters are selected to subscribe to

### CODE REVIEW

File: EngineeringModel/Services/IParameterSubscriptionBatchService .cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        /// <param name="parameterTypes">
+        /// An <see cref="IEnumerable{ParameterType}"/> that is a selection criteria to select the <see cref="Parameter"/>s to which
+        /// <see cref="ParameterSubscription"/>s need to be created
+        /// </param>
+        /// <returns>
+        /// an awaitable <see cref="Task"/>
+        /// </returns>
+        Task Create(ISession session, Iteration iteration, bool isUncategorizedIncluded, IEnumerable<Category> categories, IEnumerable<DomainOfExpertise> domainOfExpertises, IEnumerable<ParameterType> parameterTypes);
+    }
+}
```
Comment (lxatrhea) [inline EngineeringModel/Services/IParameterSubscriptionBatchService .cs:72]: newline

**Review by lxatrhea — APPROVED** [review]
One newline. Don't let that hold you from landing.

File: EngineeringModel/Services/ParameterSubscriptionBatchService.cs [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+            }
+
+            if (iteration == null)
+            {
+                throw new ArgumentNullException(nameof(iteration), $"The {nameof(iteration)} may not be null");
+            }
+
+            if (parameterTypes == null)
+            {
+                throw new ArgumentNullException(nameof(iteration), $"The {nameof(parameterTypes)} may not be null");
```
Comment (ahmedatrhea) [inline EngineeringModel/Services/ParameterSubscriptionBatchService.cs:90]: `throw new nameof(iteration) > nameof(parameterTypes)`

File: EngineeringModel/Services/ParameterSubscriptionBatchService.cs [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+            }
+
+            if (categories == null)
+            {
+                throw new ArgumentNullException(nameof(iteration), $"The {nameof(categories)} may not be null");
```
Comment (ahmedatrhea) [inline EngineeringModel/Services/ParameterSubscriptionBatchService.cs:95]: throw new nameof(iteration) > nameof(categories )

File: EngineeringModel/Services/ParameterSubscriptionBatchService.cs [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+            }
+
+            if (domainOfExpertises == null)
+            {
+                throw new ArgumentNullException(nameof(iteration), $"The {nameof(categories)} may not be null");
```
Comment (ahmedatrhea) [inline EngineeringModel/Services/ParameterSubscriptionBatchService.cs:100]: throw new nameof(iteration) > nameof(domainOfExpertises )

**Comment by ahmedatrhea** [comment]
There's a common typo in multiple places (files names, classes name and method names) where we have Paramet**eter*** instead of Parameter*. Example: [EngineeringModel/ViewModels/Dialogs/ParameteterSubscriptionFilterSelectionResult.cs ](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/blob/353b76212521265ad47ac091553279219b88bf30/EngineeringModel/ViewModels/Dialogs/ParameteterSubscriptionFilterSelectionResult.cs)

**Review by ahmedatrhea — APPROVED** [review]

**Pull request closed without merging**